runbotics/runbotics-scheduler/src/scheduler-database/credential/credential.service.ts

```diff
@@ -0,0 +1,247 @@
+
+    if (!template) {
+      throw new NotFoundException(`Could not find template with id ${credentialDto.templateId}`);
+    }
+
+    const tenantId = user.tenantId;
+
+    const attributes = await Promise.all(
+      template.attributes.map(async (attribute) => {
+        const encryptedValue = this.secretService.encrypt('', tenantId);
+        const secretEntity: Secret = {
+          ...encryptedValue,
+          tenantId,
+        };
+
+        const secret = await this.secretService.save(secretEntity)
+          .catch((error) => {
+            throw new BadRequestException(`Failed to save secret: ${error.message}`);
+          });
+
+        const secretId = secret.id;
+
+        return {
+          ...attribute,
+          secretId: secretId,
+          tenantId,
+          masked: true,
+        };
+      }));
+
+    const credential = this.credentialRepo.create({
+      ...credentialDto,
+      createdById: user.id,
+      updatedById: user.id,
+      tenantId,
+      collectionId,
+      attributes,
+    });
+
+    return this.credentialRepo.save(credential)
+      .catch(async (error) => {
+        await this.validateName(credentialDto.name, user.id);
+        throw new BadRequestException(error.message);
+      });
+  }
+
+  async findAllAccessible(tenantId: string, user: IUser) {
+    const accessibleCollections = await this.collectionService.findAll(user);
+
+    if (accessibleCollections.length === 0) {
+      throw new NotFoundException('Could not find any accessible collections');
+    }
+    return accessibleCollections.map((collection) => collection.credentials);
+
+    const collectionIds = accessibleCollections.map((collection) => collection.id);
+
+    const credentials = await this.credentialRepo.find({
+      where: {
+        tenantId,
+        collectionId: In(collectionIds),
+      },
+      relations
+    });
+
+    return credentials;
+  }
+
+  async findByCriteria(tenantId: string, criteria: Partial<Credential>) {
+    return this.credentialRepo.find({
+      where: {
+        tenantId,
+        ...criteria
+      },
+      relations
+    });
+  }
+
+  async findAllAccessibleByCollectionId(user: IUser, collectionId: string) {
+    const collection = await this.collectionService.findOneById(collectionId, user);
+
+    if (!collection) {
+      throw new NotFoundException(`Could not find collection with id ${collection}`);
+    }
+
+    const credentials = await this.findByCriteria(user.tenantId, { collectionId });
+
+    if (credentials.length === 0) {
+      throw new NotFoundException(`Could not find credentials for collection with id ${collectionId}`);
+    }
+
+    return credentials;
+  }
+
+  async findOneAccessibleById(id: string) {
+    const result = await this.credentialRepo.findOne({
+      where: {
+        id
+      },
+      relations
+    });
+
+    if (!result) {
+      throw new NotFoundException(`Could not find credential with id ${id}`);
+    }
+
+    return result;
+  }
```

Comment:
I am not sure - why we would need enpoint where there is not tenant check?
Also - for the consistence I would rename result to credential, but this is just a suggestion.